### PR TITLE
Add CI for the front-end

### PR DESCRIPTION
**Pull request by jhc-at-home**
Adds CI integration for the client side of the application.

### CODE REVIEW

File: src/components/AppHeader.jsx [hunk 1]
```diff
@@ -400,19 +401,19 @@ class AppHeader extends Component {
...
         );
         let d = pako.deflate(j, { to: 'string' });
         let b = encodeURIComponent(window.btoa(d).replace(/\//g, '$'));
-        if (this.state.permalinkUUID) {
-            var uuid = 'permalink__' + match.params.uuid;
-        } else {
-            var uuid = 'permalink';
-        }
+        const uuid =
+            'permalink' + (this.state.permalinkUUID ? match.params.uuid : '');
```
Comment (KrisDavie) [inline src/components/AppHeader.jsx]: This needs the same change bringing over from #280

Comment (jhc-at-home) [reply src/components/AppHeader.jsx]: Indeed, once #280 is merged I'll rebase on master bringing this over.

**Comment by jhc-at-home** [comment]
> Seems to be a lot of the same changes as #280? Regardless, seems to be good to me after the permalink fix

Indeed. It's intended to be rebased once #280 is merged.

**Comment by jhc-at-home** [comment]
FYI for future. PRs marked Work In Progress [WIP] needn't be reviewed yet. I'll unmark WIP and ping you when they're ready. Actually maybe to save you the hassle I just won't set reviewers until they're ready.

**Comment by KrisDavie** [comment]
> FYI for future. PRs marked Work In Progress [WIP] needn't be reviewed yet. I'll unmark WIP and ping you when they're ready. Actually maybe to save you the hassle I just won't set reviewers until they're ready.

Ah, not going to lie, didn't even see the WIP, I just glanced right over it. Saw I was requested and so just went right ahead xD

**Comment by jhc-at-home** [comment]
Ah, that means it's best I don't add reviewers until it's ready. Sorry

**Comment by jhc-at-home** [comment]
Indeed. Codefactor only uses eslint config in the master branch

**Comment by jhc-at-home** [comment]
Ping @KrisDavie @dweemx ready for review

**Comment by jhc-at-home** [comment]
Hmm... seems like a lot of noise. Can you confirm that `master` doesn't have these issues?

**Comment by dweemx** [comment]
> Hmm... seems like a lot of noise. Can you confirm that `master` doesn't have these issues?
> 
> @dweemx Could you give me _full_ reproduction steps? Which gene? What did you type? What data did you upload? Which tab were you on? Which images did you expect to see?

1. Loading any loom file e.g.: Aerts_Fly_AdultBrain_Filtered_57k.loom
2. Go to Gene tab (even Regulon tab does not work)
3. Query a gene (e.g.: onecut)

Got more descriptive error:
image

**Comment by jhc-at-home** [comment]
OK I observe that too. Unfortunately for us that file wasn't changed in this PR. Do you observe the same behaviour in `master`?

**Comment by jhc-at-home** [comment]
OK I see a potential issue in b855e9d13adb4

**Comment by dweemx** [comment]
> OK I observe that too. Unfortunately for us that file wasn't changed in this PR. Do you observe the same behaviour in `master`?

Yes

**Comment by jhc-at-home** [comment]
OK I'm also going to add a lint error for usage of `var`. God I hate javascript.

**Comment by jhc-at-home** [comment]
@dweemx @KrisDavie ready for you to have a look at again

**Comment by dweemx** [comment]
> @dweemx @KrisDavie ready for you to have a look at again

Tested locally :heavy_check_mark:

**Comment by jhc-at-home** [comment]
Sorry, just fixed another regression

**Pull request merged**